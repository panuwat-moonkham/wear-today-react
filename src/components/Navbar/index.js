import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import useScroll from '../../utils/useScroll'
import logoImage from '../../assets/logo.png'
import BackgroungBlock from '../BackgroundBlock'

const Container = styled.div`
  width: 100%;
  position: fixed;
  z-index: 10000;
  background-color: ${props => (props.isScroll ? '#fff' : 'transparent')};
  color: ${props => (props.isScroll ? '#000' : '#fff')};
  transition: background-color 0.5s;
`
const Wrapper = styled.div`
  display: flex;
  flex: 1;
  align-items: center;
  padding: 1rem;
  justify-content: space-between;
`

const ActionContainer = styled.ul`
  display: flex;
  flex: 1;
  justify-content: flex-end;
  align-items: center;
  transition: 0.3s;
`

const Action = styled.li`
  padding: 0 1rem;
`
const Login = styled.div``

const Logo = styled.div`
  width: 6%;
`

function NavBar() {
  const isScroll = useScroll({ scrollRange: 20 })
  return (
    <Container isScroll={isScroll}>
      <Wrapper>
        <Logo>
          <BackgroungBlock src={logoImage} height="8" />
        </Logo>
        <ActionContainer>
          <Action>
            <Link to="/">HOME</Link>
          </Action>
          <Action>
            <Link to="/post">POST</Link>
          </Action>
          <Action>
            <Link to="/read">READ</Link>
          </Action>
          <Action>
            <Link to="/help">HELP</Link>
          </Action>
          <Action>
            <Link to="/about">About</Link>
          </Action>
          <Action>
            <Link to="/contact">Contact</Link>
          </Action>
        </ActionContainer>
        <Login>*ปุ่มlogin*</Login>
      </Wrapper>
    </Container>
  )
}

export default NavBar
