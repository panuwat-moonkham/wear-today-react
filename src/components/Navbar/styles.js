import styled from 'styled-components'

export const Container = styled.div`
width: 100%;
position: fixed;
z-index: 10000;
background-color: ${props => (props.isScroll ? '#f1b39c' : 'transparent')};
color: ${props => (props.isScroll ? '#212121' : '#fff')};
transition: background-color 0.5s;
`
export const Wrapper = styled.div`
display: flex;
flex: 1;
align-items: center;
padding: 1rem;
`

export const ActionContainer = styled.ul`
display: flex;
flex: 1;
justify-content: flex-end;
align-items: center;
transition: 0.3s;
`

export const Action = styled.li`
padding: 0 1rem;
`
export const Login = styled.div``

export const Logo = styled.div`
width: 6%;
`

export const DropdownContent = styled.div`
display: none;
position: absolute;
background-color: #f1f1f1;
min-width: 110px;
z-index: 1;
border-radius: 5px;


a {
  color: black;
  padding: 14px 0;
  padding-left: 10px;
  text-decoration: none;
  display: block;
}
a:hover {
  background-color: #ddd;
  border-radius: 5px;
}
`

export const Dropbtn = styled.div`
border: none;

`

export const Dropdown = styled.div`
position: relative;
display: inline-block;


&:hover ${DropdownContent} {
  display: block;
}
`