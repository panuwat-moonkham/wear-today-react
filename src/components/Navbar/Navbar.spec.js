//global describe
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import Navbar from '.'

describe('Navbar', () => {
  test('should render Logo layout correcyly', () => {
    const { Logo } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(Logo).toMatchSnapshot()
  })

  test('should have link navigate to home', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(getByText(/home/gi)).toBeInTheDocument()
    expect(getByText(/home/gi).closest('a')).toHaveAttribute('href', '/')
  })
  test('should have link navigate to post', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(getByText(/post/gi)).toBeInTheDocument()
    expect(getByText(/post/gi).closest('a')).toHaveAttribute('href', '/post')
  })
  test('should have link navigate to about', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(getByText(/about/gi)).toBeInTheDocument()
expect(getByText(/about/gi).closest('a')).toHaveAttribute('href', '/about')
  })
  test('should have link navigate to contact', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(getByText(/contact/gi)).toBeInTheDocument()
expect(getByText(/contact/gi).closest('a')).toHaveAttribute('href','/contact')
  })
  test('should have link navigate to help', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(getByText(/help/gi)).toBeInTheDocument()
expect(getByText(/help/gi).closest('a')).toHaveAttribute('href', '/help')
  })

  test('should render layout correcyly', () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })
})