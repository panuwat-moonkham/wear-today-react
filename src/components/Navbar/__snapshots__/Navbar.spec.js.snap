// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render Logo layout correcyly 1`] = `undefined`;

exports[`Navbar should render layout correcyly 1`] = `
<div>
  <div
    class="sc-bdnylx eFvfMq"
  >
    <div
      class="sc-gtssRu mUgFr"
    >
      <div
        class="sc-jSFkmK eIAFAH"
      >
        <a
          href="/"
        >
          <div
            class="sc-pNWxx cvPoBq"
            height="8"
            src="logo.png"
            width="100"
          >
            <div
              class="sc-jrsJCI dtMWiw"
            />
          </div>
        </a>
      </div>
      <ul
        class="sc-dlnjPT cMDBow"
      >
        <li
          class="sc-hKFyIo fNhXog"
        >
          <a
            href="/"
          >
            HOME
          </a>
        </li>
        <li
          class="sc-hKFyIo fNhXog"
        >
          <a
            href="/post"
          >
            POST
          </a>
        </li>
        <li
          class="sc-hKFyIo fNhXog"
        >
          <a
            href="/read"
          >
            READ
          </a>
        </li>
        <li
          class="sc-hKFyIo fNhXog"
        >
          <div
            class="sc-fujyUd dOCgor"
          >
            <div
              class="sc-iCoHVE fhoggF"
            >
              SERVICE
            </div>
            <div
              class="sc-gKAblj dzXQzQ"
            >
              <a
                href="/help"
              >
                Help
              </a>
              <a
                href="/contact"
              >
                Contact
              </a>
              <a
                href="/about"
              >
                About us
              </a>
            </div>
          </div>
        </li>
        <li
          class="sc-hKFyIo fNhXog"
        >
          <a
            href="/login"
          >
            LOGIN
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
